### PR TITLE
fix(FEC-7467): error handling

### PR DESCRIPTION
**Pull request by odedhutzler**
### Description of the Changes

add and map the errors to the errors hlsjs is firing

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment 
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/hls-adapter.js [hunk 1]
```diff
@@ -449,50 +450,117 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
...
+  _handleFatalError(severity: number, category: number, code: number, description: any, callback: any): void{
+    const message = PlayerError.createError({
+      severity: severity,
+      category: category,
+      code: code,
+      args: {data: description}
+    });
+    this._trigger(BaseMediaSourceAdapter.CustomEvents.ERROR,message);
+    HlsAdapter._logger.error(description);
+    callback && callback();
```
Comment (OrenMe) [inline src/hls-adapter.js]: No reason to mix the error dispatching and error handing, better handle the "callback" logic where it is already handled now

File: src/hls-adapter.js [hunk 2]
```diff
@@ -449,50 +450,117 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
...
     let errorDetails = data.details;
     let errorFatal = data.fatal;
+    let description = "";
     if (errorFatal) {
       switch (errorType) {
         case Hlsjs.ErrorTypes.NETWORK_ERROR:
-          HlsAdapter._logger.error("fatal network error encountered, try to recover");
-          this._hls.startLoad();
+          description = "fatal network error encountered, try to recover";
+          this._handleFatalError(PlayerError.Severity.RECOVERABLE, PlayerError.Category.NETWORK, PlayerError.Category.HTTP_ERROR, description, this._hls.startLoad);
```
Comment (OrenMe) [inline src/hls-adapter.js]: sending a `_handleFatalError` a RECOVERABLE? kind of unintuitive...

File: src/hls-adapter.js [hunk 3]
```diff
@@ -449,50 +450,117 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
...
           break;
         case Hlsjs.ErrorTypes.MEDIA_ERROR:
-          HlsAdapter._logger.error("fatal media error encountered, try to recover");
-          this._hls.recoverMediaError();
+          description = "fatal media error encountered, try to recover";
+          this._handleFatalError(PlayerError.Severity.RECOVERABLE, PlayerError.Category.MEDIA, PlayerError.Category.HLS_FATAL_MEDIA_ERROR, description, this._hls.recoverMediaError);
```
Comment (OrenMe) [inline src/hls-adapter.js]: sending a `_handleFatalError` a RECOVERABLE? kind of unintuitive...

File: src/hls-adapter.js (hunk 1 above)
Comment (odedhutzler) [reply src/hls-adapter.js]: K

File: src/hls-adapter.js [hunk 4]
```diff
@@ -132,14 +142,15 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
...
+    this._hls.on(Hlsjs.Events.ERROR, (e) => {
+      this._onError(e);});
     this._hls.on(Hlsjs.Events.MANIFEST_LOADED, this._onManifestLoaded.bind(this));
     this._hls.on(Hlsjs.Events.LEVEL_SWITCHED, this._onLevelSwitched.bind(this));
     this._hls.on(Hlsjs.Events.AUDIO_TRACK_SWITCHED, this._onAudioTrackSwitched.bind(this));
   }
 
   /**
-   * Load the video source
+   * Load the video source need moew
```
Comment (OrenMe) [inline src/hls-adapter.js]: go cats!

**Pull request merged**